text/0000-rustdoc-url-conflicts.md

```diff
@@ -0,0 +1,65 @@
+- Feature Name: rustdoc_url_conflicts
+- Start Date: 2021-03-29
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC describes how the rustdoc URLs will be modified to fix the current issue on case insensitive file systems.
+For example, in the libstd there is:
+ * keyword.self.html
+ * keyword.Self.html
+
+If macOS or Windows users try to access to both URLs, they will always see the same content because for both file systems, it's the same file.
+
+# Motivation
+[motivation]: #motivation
+
+This is one of the oldest rustdoc issue and has been problematic for years and a lot of users.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The idea here is to replace every capitalized letter in the item with `-[minimized capital]`. So for example, `enum.FooBar.html` will become `enum.-foo-bar.html`.
```

Comment:
It seems really unfortunate that this would make even incredibly common things like `struct.Vec.html` have URLs that are material worse (in my opinion).

Could it at least use a slightly more complicated scheme to take advantage of the usual rust naming conventions?  Like have `struct vec` be the one that gets `struct.-vec.html`, since it's normal for `struct`s to start with an upper-case letter.

Reply:
I agree, I don't think it's good that _common_ types get affected by this

Reply:
That wouldn't help with cases where the case differs mid-identifier and could be valid either way `struct FooBar; struct FoObar;`.

Reply:
I don't understand your point @Nemo157. It would be `-fo-obar`. It would prevent name conflict, which is the primary issue here.

Reply:
I was referring to @scottmcm's alternative.

Reply:
Oh my bad. Then yes, I completely agree with you. :)

Reply:
@Nemo157 Yes, that's not a complete fix, as it'd still change `struct.FooBar.html` to `struct.Foo-Bar.html` or similar.

But I don't think we should break non-agglutinative names just because we have to break agglutinative ones.

Reply:
No, I guess my explanations are really bad... All capital letters are *replaced* by a dash followed by the lowercase version. So `FooBar` will become `struct.-foo-bar.html`. With this, no more conflicts.